### PR TITLE
blueprints: Use minimal ".eslintrc" config files

### PR DESCRIPTION
**Pull request by Turbo87**
This PR replaces the `eslint-config-ember` dependency with a minimal `.eslintrc` file based on the built-in `eslint:recommended` ruleset as discussed in #43.

Note that the testing `.eslintrc` file will not work properly yet until `broccoli-lint-eslint` is updated to support inherited config files.

/cc @alexlafroscia @mmun

### CODE REVIEW

File: blueprints/ember-cli-eslint/files/tests/.eslintrc [hunk 1]
```diff
@@ -1,6 +1,5 @@
 {
-  "extends": "ember",
-  "rules": {
-    "no-console": 0
+  "env": {
+    "embertest": true
```
Comment (alexlafroscia) [inline blueprints/ember-cli-eslint/files/tests/.eslintrc]: I'm curious, what does this do?

Comment (Turbo87) [inline blueprints/ember-cli-eslint/files/tests/.eslintrc]: it's allowing [these](https://github.com/sindresorhus/globals/blob/master/globals.json#L1199-L1212) globals

Comment (alexlafroscia) [inline blueprints/ember-cli-eslint/files/tests/.eslintrc]: Oh, very cool! I'd never heard of that before.

**Comment by mmun** [comment]
@Turbo87 Does this configuration support async/await? The default parser (espree) doesn't seem to understand it. At least according to http://eslint.org/parser/.

**Comment by Turbo87** [comment]
@mmun I don't know, but since async/await is unfortunately not stable yet and not widely used either I would prefer to skip the babel-eslint dependency for now. just like the custom config this is something that can be added manually if the app/addon is actually needing it.

**Comment by morhook** [comment]
:+1: to this new generated files, it will give a better `user-experience` for people installing the plugin (and not using `babel-eslint`).

**Comment by mmun** [comment]
@Turbo87 Ah I see. Adding the extra dependency would be annoying indeed...

**Comment by alexlafroscia** [comment]
Decorators end up in the same boat as `async/await`; some people are starting to use them but they make the default parser choke.

I personally like the idea of using `babel-eslint` by default.  That way, the code is parsed only with Babel, and you're never left wondering if some feature is supported by the linter too.  I've been on a project using JSHint that had that frustration, since JSHint didn't support something that Babel allowed.  As a counter-point, however, `babel-eslint` is pretty easy to install if that's what you need.

If there's no good way to manage that extra dependency as a sub-dependency of `ember-cli-eslint` then it's probably better to just leave it out, but I think using it by default, if there's a clean way to do so, makes a nicer user experience overall.

**Comment by Turbo87** [comment]
@alexlafroscia I think we would need to add that dependency to the user projects via blueprint since the parser won't be found by ESLint otherwise, but as you say this is very easy to do manually and if pointed out in the README of this addon it is pretty much a no-brainer.

**Comment by alexlafroscia** [comment]
@Turbo87 Yeah, that's a better approach then.  Personally, I hate when addons install their own NPM/Bower dependencies.  Too bad it can't all be contained within the addon itself.

**Comment by mmun** [comment]
It may even be possible to make a sort of "plugin" addon `ember-cli-eslint-babel` that depends on `babel-eslint` and has some helpful blueprints for you. It depends on how `eslint` resolves the parser exactly.

**Comment by mmun** [comment]
If https://github.com/jonathanKingston/broccoli-lint-eslint/issues/33 gets resolved this gives us a ton of flexibility. We could reduce the boilerplate down to

```js
module.exports = require('ember-cli-eslint').config({
  rules: {}
});
```

by putting all the defaults into `require('ember-cli-eslint').config`. At that point it would be much easier to get `babel-eslint` as the default parser. We should just need to have `ember-cli-eslint` depend on `babel-eslint` and pass its resolved path through to eslint's `parser` configuration property.

**Comment by alexlafroscia** [comment]
@mmun I was just about to say that it should be possible to provide the parser through `ember-cli-eslint`, through `broccoli-lint-eslint` right to ESLint itself.  

> We should just need to have ember-cli-eslint depend on babel-eslint and pass its resolved path through to eslint's parser configuration property.

If we could do something like this, I feel like it might be worth it, so that no one has to even think about the parser.  If there's no downside to using `babel-eslint` as the parser, other than having to install it and update the configuration, I think it would be easier to use the plugin if that was the default.

**Comment by mmun** [comment]
@alexlafroscia Yes, we already do a similar thing in ember-cli-babel and ember-cli-qunit

https://github.com/babel/ember-cli-babel/blob/master/index.js#L39-L41
https://github.com/ember-cli/ember-cli-qunit/blob/865729c0c8aae2cbb49df63702f7800b9312bf2a/index.js#L19-L20

**Comment by alexlafroscia** [comment]
@mmun Awesome! Thanks for those links.

**Comment by Turbo87** [comment]
I've rebased the PR and changed from `.eslintrc` to `.eslintrc.js` files

**Comment by BrianSipple** [comment]
Apologies for chiming in a bit late, but I like a lot of the directions that appear to have been laid out here.

Regarding `babel-eslint`, I’ve always felt that the current installation instruction at the top of the README was a perfectly intuitive, super straightforward way to instruct users about the potential need for it without making it a hard dependency. I think it would be good to stick with that approach for now — but also be on the lookout for feedback going forward.

Also, props for the lightweight default ruleset. I remember chatting with @rwjblue a few weeks ago about how the current settings were pretty cumbersome, and so this feels like a nice clean slate.

That being said, I’m still intrigued by the _idea_ that `eslint-config-ember` advertised. Its trouble was that it wasn’t really eslint-config-_ember_ per se, but rather a few patchwork rules on top of an entirely different (and rather strict) project setup. However, I think having a community-curated set of rules that aligned with, say, `ember-cli` coding conventions could be highly useful to a lot of people — and it wouldn’t even have to be a strict dependency here, but rather it could just be something that we maintained in the ember-cli organization and then referenced as a suggested extension in the README.

Anyway, that last point's more food for thought than anything. Everything in the PR here LGTM.

**Comment by jonathanKingston** [comment]
Why not replace: `eslint-config-ember` It's the communities etc, I can move it over and move all ember rules there. It's a little cleaner that way?

**Comment by BrianSipple** [comment]
This is a great discussion, but everything here in the PR seems like a pure win. @Turbo87, @mmun, and @alexlafroscia, would you all be okay with merging what's here and opening up a new issue regarding the [changes that were proposed above](https://github.com/ember-cli/ember-cli-eslint/pull/53#issuecomment-209601316) (and perhaps another issue regarding base configurations, `eslint-config-ember`, etc)?

**Comment by Turbo87** [comment]
:+1:

**Pull request merged**